kernel/ps2/src/lib.rs

```diff
@@ -1,484 +1,704 @@
+    /// returns [PortTestResult]
+    TestPort1 = 0xAB,
+    ///// read all bytes of internal RAM
+    // DiagnosticDump = 0xAC,
+    /// sets [ControllerConfigurationByte] `port1_clock_disabled`
+    DisablePort1 = 0xAD,
+    /// clears [ControllerConfigurationByte] `port1_clock_disabled`
+    EnablePort1 = 0xAE,
+    // ReadControllerInputPort = 0xC0, //return Unknown/non-standard
+    // CopyBits0to3ofInputPortToStatusBits4to7 = 0xC1,
+    // CopyBits4to7ofInputPortToStatusBits4to7 = 0xC2,
+    ///// returns [ControllerOutputPort] (not implemented, https://wiki.osdev.org/%228042%22_PS/2_Controller#PS.2F2_Controller_Output_Port)
+    // ReadControllerOutputPort = 0xD0,
+    ///// reads [ControllerOutputPort] (not implemented, https://wiki.osdev.org/%228042%22_PS/2_Controller#PS.2F2_Controller_Output_Port)
+    // WriteByteToControllerOutputPort = 0xD1, //Check if output buffer is empty first
+    ///// makes it look like the byte written was received from the first PS/2 port
+    // WriteByteToPort1OutputBuffer = 0xD2, //NOTE: only if 2 PS/2 ports supported
+    ///// makes it look like the byte written was received from the second PS/2 port
+    // WriteByteToPort2OutputBuffer = 0xD3, //NOTE: only if 2 PS/2 ports supported //https://wiki.osdev.org/%228042%22_PS/2_Controller#Buffer_Naming_Perspective OUTPUT means "out of the device into host"
+    /// sends next byte to the second PS/2 port
+    WriteByteToPort2InputBuffer = 0xD4, //NOTE: only if 2 PS/2 ports supported
+    // ReadTestInputs = 0xE0,
+    // PulseOutputLineLowFor6ms = 0xF0, //0xF0-0xFF; Bits 0 to 3 correspond to 4 different output lines and are used as a mask (0 = pulse line, 1 = don't pulse line); Bit 0 corresponds to the "reset" line. The other output lines don't have a standard/defined purpose.
+}
 
-static PS2_PORT: Mutex<Port<u8>> = Mutex::new(Port::new(0x60));
-static PS2_COMMAND_PORT: Mutex<Port<u8>> = Mutex::new(Port::new(0x64));
-
-/// clean the PS2 data port (0x60) output buffer
-/// also return the vec of data previously in the buffer which may be useful
-pub fn ps2_clean_buffer() -> Vec<u8> {
-    let mut buffer_data = Vec::new();
-    loop {
-        // if no more data in the output buffer
-        if ps2_status_register() & 0x01 == 0 {
-            break;
-        } else {
-            buffer_data.push(ps2_read_data());
-        }
+/// Write a command to the PS/2 command port/register
+/// Note: Devices attached to the controller should be disabled
+/// before sending commands that return data, otherwise the output buffer could get overwritten
+fn write_command(value: HostToControllerCommand) {
+    unsafe {
+        PS2_COMMAND_AND_STATUS_PORT.lock().write(value as u8);
     }
-    buffer_data
 }
 
-/// write command to the command ps2 port (0x64)
-pub fn ps2_write_command(value: u8) {
-    unsafe {
-        PS2_COMMAND_PORT.lock().write(value);
+#[allow(non_camel_case_types)]
+type u1 = B1;
+#[allow(non_camel_case_types)]
+type bool1 = bool;
```

Comment:
I think I will just remove these, I kind of wanted to emulate what the above mentioned arbitrary bit-width integer people proposed, but if that ever hits rust it would be a simple search & replace anyways.